packages/tools/src/eventDispatchers/shared/getActiveToolForTouchEvent.ts

```diff
@@ -0,0 +1,59 @@
+import { ToolGroupManager } from '../../store';
+import { MouseBindings, ToolModes } from '../../enums';
+import { EventTypes } from '../../types';
+import getMouseModifier from './getMouseModifier';
+
+const { Active } = ToolModes;
+
+/**
+ * Iterate tool group tools until we find a tool that has a "ToolBinding"
+ * that matches our TouchEvent's `buttons`. It's possible there will be no match
+ * (no active tool for that touch button combination).
+ *
+ * @param evt - The event dispatcher touch event.
+ *
+ * @returns tool
+ */
+export default function getActiveToolForTouchEvent(
+  evt: EventTypes.NormalizedTouchEventType
+) {
+  // Todo: we should refactor this to use getToolsWithModesForTouchEvent instead
+  const { renderingEngineId, viewportId } = evt.detail;
+  const touchEvent = evt.detail.event;
+
+  const toolGroup = ToolGroupManager.getToolGroupForViewport(
+    viewportId,
+    renderingEngineId
+  );
+
+  if (!toolGroup) {
+    return null;
+  }
+
+  const toolGroupToolNames = Object.keys(toolGroup.toolOptions);
+
+  const numTouchPoints = Object.keys(touchEvent.touches).length;
+
+  // If any keyboard modifier key is also pressed
+  const modifierKey = getMouseModifier(touchEvent);
+
+  for (let j = 0; j < toolGroupToolNames.length; j++) {
+    const toolName = toolGroupToolNames[j];
+    const toolOptions = toolGroup.toolOptions[toolName];
+
```

Comment:
I'm confused about the numTouchPoints === 1 and mix of it here with mouse button. Can you please clarify? and also mention it in the doc for touch? what is the relation of mouse and touch for the case of numTouchPoints === 1. I think it is a special case?

Reply:
Yeah numTouchPoints = 1 is a special case where we also treat it as a "Primary Mouse" event. This way existing modes which just bind a primary mouse get the free touch win without additional explicit config.

Reply:
We can make a note in the doc